### PR TITLE
Support Alpine OS

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
***Description:*** Starting to support the agent for Alpine x64

* `src/Agent.Worker/ContainerOperationProvider.cs` —
  — commands `useradd`, `groupadd`, and `usermod` were changed for Alpine
  (this needs to be released before releasing the agent for Alpine)

* `src/Agent.Worker/Handlers/NodeHandler.cs` —
  — using node V10 instead of node V6 (node V6 does not exist on Alpine)

* `src/dev.sh` — `RuntimeIdentifier` for Alpine is set to `linux-musl-x64` — dotnet SDK will be downloaded from
  https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.405/dotnet-sdk-6.0.405-linux-musl-x64.tar.gz

* `src/Misc/externals.sh` —
  — URL to download node for Alpine is set to
  https://unofficial-builds.nodejs.org/download/release `/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64-musl.tar.gz`

* Building the agent for Alpine is added to the agent CI
  (currently disabled since `ContainerOperationProvider` needs to be updated first)

* Unit tests are updated

* Checked that applied changes work as expected (tested end-to-end)

### CODE REVIEW

**Review by vmapetr — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

File: src/Agent.Worker/ContainerOperationProvider.cs [hunk 1]
```diff
@@ -606,20 +606,54 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
...
+                    {
+                        await DockerExec(executionContext, container.ContainerId, "useradd");
+
+                        addGroup = (groupName) => $"groupadd {groupName}";
+                        addGroupWithId = (groupName, groupId) => $"groupadd -g {groupId} {groupName}";
+                        addUserWithId = (userName, userId) => $"useradd -m -u {userId} {userName}";
+                        addUserWithIdAndGroup = (userName, userId, groupName) => $"useradd -m -g {groupName} -u {userId} {userName}";
+                        addUserToGroup = (userName, groupName) => $"usermod -a -G {groupName} {userName}";
+                    }
+                    catch (Exception ex) when (ex is InvalidOperationException)
```
Comment (merlynomsft) [inline src/Agent.Worker/ContainerOperationProvider.cs:626]: > catch (Exception ex) when (ex is InvalidOperationException) [](http://example.com/codeflow?start=19&length=60)

Just catch(InvalidOperation exception), no need for pattern matching here, unless I'm misisng something

File: src/Agent.Worker/ContainerOperationProvider.cs [hunk 2]
```diff
@@ -606,20 +606,54 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
...
-                    if (string.IsNullOrEmpty(containerUserName)) // Create a new user with same UID as on the host
+                    // List of commands
+                    Func<string, string> addGroup;
+                    Func<string, string, string> addGroupWithId;
+                    Func<string, string, string> addUserWithId;
+                    Func<string, string, string, string> addUserWithIdAndGroup;
+                    Func<string, string, string> addUserToGroup;
+
+                    try
+                    {
```
Comment (merlynomsft) [inline src/Agent.Worker/ContainerOperationProvider.cs:617]: exception handler seems broad, please narrow it exactly where InvalidOperation is expected to be thrown.  (probably the DockerExec)

File: src/Agent.Sdk/Util/PlatformUtil.cs [hunk 3]
```diff
@@ -74,6 +74,26 @@ public static bool RunningOnLinux
...
+            get
+            {
+                try
+                {
+                    if (File.Exists("/etc/alpine-release"))
+                    {
+                        return true;
+                    }
+                }
+                catch (Exception)
```
Comment (merlynomsft) [inline src/Agent.Sdk/Util/PlatformUtil.cs:88]: >Exception [](http://example.com/codeflow?start=23&length=9)

Please narrow this to the exception you're expecting here.  If not expecting an exception, remove try/catch block

File: src/Agent.Worker/ContainerOperationProvider.cs (hunk 2 above)
Comment (DenisRumyantsev) [reply src/Agent.Worker/ContainerOperationProvider.cs]: Fixed, thanks!

File: src/Agent.Sdk/Util/PlatformUtil.cs (hunk 3 above)
Comment (DenisRumyantsev) [reply src/Agent.Sdk/Util/PlatformUtil.cs]: No exceptions are expected here. Fixed, thanks!

File: src/Agent.Worker/Handlers/NodeHandler.cs [hunk 4]
```diff
@@ -225,12 +225,8 @@ public string GetNodeLocation()
             string useNodeKnob = AgentKnobs.UseNode.GetValue(ExecutionContext).AsString();
 
             string nodeFolder = NodeHandler.nodeFolder;
-            if (PlatformUtil.RunningOnRHEL6 && taskHasNode16Data)
```
Comment (merlynomsft) [inline src/Agent.Worker/Handlers/NodeHandler.cs:228]: >RunningOnRHEL6 [](http://example.com/codeflow?start=29&length=14)

not sure if it matters but we lost the reference to RunningOnRHEL6  in the changes

**Review by merlynomsft — APPROVED** [review]
:shipit:

File: src/Agent.Worker/Handlers/NodeHandler.cs (hunk 4 above)
Comment (DenisRumyantsev) [reply src/Agent.Worker/Handlers/NodeHandler.cs:228]: We officially dropped RHEL6 support, so all RHEL6-related logic should be cut out completely (probably not in this PR).

**Pull request merged**